native/cocos/editor-support/spine-wasm/spine-type-export.cpp

```diff
@@ -85,20 +34,467 @@ using SPVectorPathConstraintPtr = Vector<PathConstraint*>;
 using SPVectorTrackEntryPtr = Vector<TrackEntry*>;
 using SPVectorUpdatablePtr = Vector<Updatable*>;
+using SPVectorSkinEntryPtr = Vector<Skin::AttachmentMap::Entry*>;
+using SPVectorVectorSkinEntryPtr = Vector<SPVectorSkinEntryPtr>;
+using SPVectorDebugShape = Vector<SpineDebugShape>;
+
+#define DEFINE_ALLOW_RAW_POINTER(type) \
+namespace emscripten { namespace internal { \
+    template<> \
+    struct TypeID<type*> { \
+        static constexpr TYPEID get() { \
+            return TypeID<type>::get(); \
+        } \
+    }; \
+    template<> \
+    struct TypeID<const type*> { \
+        static constexpr TYPEID get() { \
+            return TypeID<type>::get(); \
+        } \
+    }; \
+}}
+
+
+#define DEFINE_SPINE_CLASS_TYPEID(cls) \
+namespace emscripten { namespace internal { \
+    template<> \
+    constexpr TYPEID getLightTypeID<cls>(const cls& value) { \
+        return value.getRTTI().getClassName(); \
+    } \
+    template<> \
+    struct LightTypeID<cls* const> { \
+        static constexpr TYPEID get() { \
+            return #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<cls*> { \
+        static constexpr TYPEID get() { \
+            return #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<cls* const &> { \
+        static constexpr TYPEID get() { \
+            return #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<cls*&> { \
+        static constexpr TYPEID get() { \
+            return #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls*> { \
+        static constexpr TYPEID get() { \
+            return "const " #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls* const> { \
+        static constexpr TYPEID get() { \
+            return "const " #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls* &> { \
+        static constexpr TYPEID get() { \
+            return "const " #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls* const &> { \
+        static constexpr TYPEID get() { \
+            return "const " #cls "*"; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<cls> { \
+        static constexpr TYPEID get() { \
+            return #cls; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<cls&> { \
+        static constexpr TYPEID get() { \
+            return #cls; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls> { \
+        static constexpr TYPEID get() { \
+            return #cls; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls&> { \
+        static constexpr TYPEID get() { \
+            return #cls; \
+        } \
+    }; \
+}}
+
+#define GETTER_RVAL_TO_PTR(ClassType, Method, ReturnType) \
+    optional_override([](ClassType &obj) { return const_cast<ReturnType>(&obj.Method()); })
```

Comment:
A easy way to make the emscripten API `.property(...)` return a pointer rather than a reference.
Returning a pointer which will not new an object and do a copy operation.